src/promptflow-azure/promptflow/azure/_utils/general.py

```diff
@@ -47,3 +59,36 @@ def set_event_loop_policy():
         # Reference: https://stackoverflow.com/questions/45600579/asyncio-event-loop-is-closed-when-getting-loop
         # On Windows seems to be a problem with EventLoopPolicy, use this snippet to work around it
         asyncio.set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())
+
+
+def is_workspace(workspace: Workspace) -> bool:
+    return workspace._kind == WorkspaceKind.DEFAULT
+
+
+def is_hub(workspace: Workspace) -> bool:
+    return workspace._kind == WorkspaceKind.HUB
+
+
+def is_project(workspace: Workspace) -> bool:
+    return workspace._kind == WorkspaceKind.PROJECT
+
+
+def _get_credential() -> typing.Union[AzureCliCredential, DefaultAzureCredential]:
```

Comment:
there is a func `get_credentials_for_cli`, seems duplicated

Reply:
good catch, has leverage that